app/Notifications/RequestAssetNotification.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace App\Notifications;
+
+use App\Models\Setting;
+use Illuminate\Bus\Queueable;
+use Illuminate\Notifications\Messages\MailMessage;
+use Illuminate\Notifications\Messages\SlackMessage;
+use Illuminate\Notifications\Notification;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class RequestAssetNotification extends Notification
+{
+    use Queueable;
+    /**
+     * @var
+     */
+    private $params;
+
+    /**
+     * Create a new notification instance.
+     *
+     * @param $params
+     */
+    public function __construct($params)
+    {
+        $this->target = $params['target'];
+        $this->item = $params['item'];
+        $this->item_type = $params['item_type'];
+        $this->item_quantity = $params['item_quantity'];
+        $this->note = '';
+        $this->last_checkout = '';
+        $this->expected_checkin = '';
+        $this->requested_date = \App\Helpers\Helper::getFormattedDateObject($params['requested_date'], 'datetime',
+            false);
+        $this->settings = Setting::getSettings();
+
+        if (array_key_exists('note', $params)) {
+            $this->note = $params['note'];
+        }
+
+        if ($this->item->last_checkout) {
+            $this->last_checkout = \App\Helpers\Helper::getFormattedDateObject($this->item->last_checkout, 'date',
+                false);
+        }
+
+        if ($this->item->expected_checkin) {
+            $this->expected_checkin = \App\Helpers\Helper::getFormattedDateObject($this->item->expected_checkin, 'date',
+                false);
+        }
+
+
+    }
+
+    /**
+     * Get the notification's delivery channels.
+     *
+     * @param  mixed  $notifiable
+     * @return array
+     */
+    public function via()
+    {
+
+        $notifyBy = [];
+
+        if (Setting::getSettings()->slack_endpoint!='') {
+            \Log::debug('use slack');
+            $notifyBy[] = 'slack';
+        }
+
+
+        $notifyBy[] = 'mail';
+
+        return $notifyBy;
```

Comment:
If this logic repeats in a lot of places - and I suspect it does - would it make sense to create a class called, like, `SwitchableNotification` which has that `via` pre-filled, and have all of your notifications derive from it?

It's a pretty dang small number of lines so it could be way more complexity for far too little payoff.

Reply:
It seemed simpler this way. Once we expand the notifications section a little more (so you can toggle checkin, checkout, etc notifications for all items, I'll the refactoring all of that. This is also where those dingleberry Recipients models come in, since people will likely want different types of notifications to go to different addresses.